packages/yew/src/suspense/suspension.rs

```diff
@@ -0,0 +1,120 @@
+use std::cell::RefCell;
+use std::future::Future;
+use std::rc::Rc;
+use std::sync::atomic::{AtomicBool, Ordering};
+
+use thiserror::Error;
+use wasm_bindgen_futures::spawn_local;
+
+use crate::Callback;
+
+thread_local! {
+    static SUSPENSION_ID: RefCell<usize> = RefCell::default();
+}
```

Comment:
Would there be a way to initialize this state only if Suspense is used in the application?

Reply:
Im kind og intrested why that is important? Its only a usize.

Reply:
It's not just a usize. There's a thread local and RefCell too. 

`therad_local!` generates code to initialize a `LocalKey` (iirc that's the type name, not 100% sure) which gets executed.  
`RefCell` isn't just a simple wrapper over usize, it's `UnsafeCell<usize>` + some other stuff.

Reply:
> Would there be a way to initialize this state only if Suspense is used in the application?

LocalKey::new is `const fn`. So there's no runtime cost until `.with` is called.